[Core] command listdisabled error in DMs

# Command bugs

<!-- 
Did you find a bug with a command? Fill out the following:
-->

#### Command name

`command listdisabled` and `command listdisabled guild` in DMs

#### What cog is this command from?

Core

#### What were you expecting to happen?

Do nothing, or announce this command doesn't work in DMs.

#### What actually happened?

```
[ERROR] red: Exception in command 'command listdisabled guild'
Traceback (most recent call last):
  File "C:\Users\Bobloy\Documents\GitHub\Red-DiscordBot\.venv\lib\site-packages\discord\ext\commands\core.py", line 85, in wrapped
    ret = await coro(*args, **kwargs)
  File "C:\Users\Bobloy\Documents\GitHub\Red-DiscordBot\redbot\core\core_commands.py", line 3846, in list_disabled_guild
    Examples:
  File "C:\Users\Bobloy\Documents\GitHub\Red-DiscordBot\redbot\core\config.py", line 976, in guild
    return self._get_base_group(self.GUILD, str(guild.id))
AttributeError: 'NoneType' object has no attribute 'id'
```

#### How can we reproduce this issue?

Be a server owner but not the bot owner and use `[p]command listdisabled` in DMs, or be bot owner and use `[p]command listdisabled guild` in DMs

--- a/redbot/core/core_commands.py
+++ b/redbot/core/core_commands.py
@@ -3199,6 +3199,7 @@ class Core(commands.commands._RuleDropper, commands.Cog, CoreLogic):
         paged[0] = header + paged[0]
         await ctx.send_interactive(paged)
 
+    @commands.guild_only()
     @list_disabled.command(name="guild")
     async def list_disabled_guild(self, ctx: commands.Context):
         """List disabled commands in this server."""

